### PR TITLE
[RDBMS] Add List-SKUS Command, Table Transformers, Local Context for Postgres, MySQL, Mariadb Single Server

### PR DESCRIPTION
**Pull request by DaeunYim**
**Description**  

1. Added list-skus commands for single server Postgres and MySQL
2. Added table transformers for list-skus and server list commands
3. Added local-context support for single server

**Testing Guide**  

1. List-skus command: run with the below commands
- az postgres server list-skus -l eastus -o table
- az mysql server list-skus -l eastus -o table
- az mariadb server list-skus -l eastus -o table

2. server list command: run with the below commands
- az postgres server list -g <group_name> -n <server_name> -o table
- az mysql server list -g <group_name> -n <server_name> -o table
- az mariadb server list -g <group_name> -n <server_name> -o table

3. local context command: 
az config param-persist on
- az postgres -h and run commands to check if server name argument is included in the local context
e.g.) az postgres server create -g <group_name> -n <server_name> -l <location> => az postgres server show

**History Notes**  

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
RDBMS

File: src/azure-cli/azure/cli/command_modules/rdbms/_transformers.py [hunk 1]
```diff
@@ -56,3 +56,38 @@ def table_transform_output_list_sku(result):
                 raise CLIError("There is no sku available for this location.")
 
     return table_result
+
+
+def table_transform_output_list_servers_single_server(result):
```
Comment (arde0708) [inline src/azure-cli/azure/cli/command_modules/rdbms/_transformers.py:61]: Nit : This is exactly same as table_transform_output_list_servers except haState and availablityZone. So try and see either

 1. If you can pass a parameter to the function and based on the parameter value( server/flexible server) , determine whether or not to add those two fields, or
 2. Have a common function and remove the repeated code there and make both these functions call the common function wherever applicable and add the rest of the exclusive fields in the respective caller.

File: src/azure-cli/azure/cli/command_modules/rdbms/_client_factory.py [hunk 2]
```diff
@@ -278,6 +278,14 @@ def cf_postgres_server_ad_administrators_operations(cli_ctx, _):
     return get_postgresql_management_client(cli_ctx).server_administrators
 
 
+def cf_mysql_location_based_performance_tier_operations(cli_ctx, _):
```
Comment (arde0708) [inline src/azure-cli/azure/cli/command_modules/rdbms/_client_factory.py:281]: Do we not have this for MariaDB ?

File: src/azure-cli/azure/cli/command_modules/rdbms/_help.py [hunk 3]
```diff
@@ -1221,3 +1221,19 @@
     type: command
     short-summary: Show the connection strings for a PostgreSQL flexible-server database.
 """
+
```
Comment (arde0708) [inline src/azure-cli/azure/cli/command_modules/rdbms/_help.py:1224]: If you add for this command for MariaDb, update this as well

File: src/azure-cli/azure/cli/command_modules/rdbms/_params.py [hunk 4]
```diff
@@ -130,10 +126,15 @@ def _complex_params(command_group):
             c.argument('file_last_written', type=int, help='Integer in hours to indicate file last modify time, default value is 72.')
             c.argument('filename_contains', help='The pattern that file name should match.')
 
+        with self.argument_context('{} server-logs list'.format(command_group)) as c:
```
Comment (arde0708) [inline src/azure-cli/azure/cli/command_modules/rdbms/_params.py:129]: I hope you have validated/ tested these changes

File: src/azure-cli/azure/cli/command_modules/rdbms/commands.py [hunk 5]
```diff
@@ -212,6 +218,18 @@ def load_command_table(self, _):
         resource_type=ResourceType.MGMT_RDBMS
     )
 
+    mysql_location_based_performance_tier_sdk = CliCommandType(
+        operations_tmpl='azure.mgmt.rdbms.mysql.operations#LocationBasedPerformanceTierOperations.{}',
+        client_factory=cf_mysql_location_based_performance_tier_operations,
+        resource_type=ResourceType.MGMT_RDBMS
+    )
+
+    postgres_location_based_performance_tier_sdk = CliCommandType(
```
Comment (arde0708) [inline src/azure-cli/azure/cli/command_modules/rdbms/commands.py:227]: Required for MariaDB if the API exists

File: src/azure-cli/azure/cli/command_modules/rdbms/_transformers.py (hunk 1 above)
Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/_transformers.py:61]: Resolved. Thanks!!

File: src/azure-cli/azure/cli/command_modules/rdbms/_help.py (hunk 3 above)
Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/_help.py:1224]: Same!

File: src/azure-cli/azure/cli/command_modules/rdbms/_help.py [hunk 6]
```diff
@@ -1221,3 +1221,27 @@
...
+  - name: Lists available sku's in the given region.
+    text: az mysql server list-skus -l eastus
+"""
+
+helps['mariadb server list-skus'] = """
+type: command
+short-summary: Lists available sku's in the given region.
+examples:
+  - name: Lists available sku's in the given region.
+    text: az mariadb server list-skus -l eastus
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/rdbms/_help.py:1246]: Lists->List

File: src/azure-cli/azure/cli/command_modules/rdbms/_params.py [hunk 7]
```diff
@@ -32,24 +32,20 @@ def load_arguments(self, _):    # pylint: disable=too-many-statements
...
-
+        server_name_help = "Name of the server. The name can contain only lowercase letters, numbers, and the hyphen (-) character. " + \
+                           "Minimum 3 characters and maximum 63 characters."
+        server_name_scope = ['{}'.format(command_group)]
+        server_name_setter_arg_type = CLIArgumentType(metavar='NAME', help=server_name_help, id_part='name',
+                                                      local_context_attribute=LocalContextAttribute(name='server_name', actions=[LocalContextAction.SET], scopes=server_name_scope))
+        server_name_getter_arg_type = CLIArgumentType(metavar='NAME', help=server_name_help, id_part='name',
+                                                      local_context_attribute=LocalContextAttribute(name='server_name', actions=[LocalContextAction.GET], scopes=server_name_scope))
+        server_name_arg_type = CLIArgumentType(metavar='NAME', help=server_name_help, id_part='name',
+                                               local_context_attribute=LocalContextAttribute(name='server_name', actions=[LocalContextAction.SET, LocalContextAction.GET], scopes=server_name_scope))
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/rdbms/_params.py:43]: @arrownj could you help with local context part?

File: src/azure-cli/azure/cli/command_modules/rdbms/commands.py [hunk 8]
```diff
@@ -458,3 +483,21 @@ def load_command_table(self, _):
...
+                            postgres_location_based_performance_tier_sdk,
+                            client_factory=cf_postgres_location_based_performance_tier_operations,
+                            is_preview=True) as g:
+        g.command('list-skus', 'list', table_transformer=table_transform_output_list_skus_single_server)
+
+    with self.command_group('mariadb server',
+                            mariadb_location_based_performance_tier_sdk,
+                            client_factory=cf_mariadb_location_based_performance_tier_operations,
+                            is_preview=True) as g:
+        g.command('list-skus', 'list', table_transformer=table_transform_output_list_skus_single_server)
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/rdbms/commands.py:503]: please add tests for all these new commands.

Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/rdbms/commands.py:503]: If there is limited api version supported, please specify min_api also.

Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/commands.py:503]: Tests including these new commands are added in this pull request https://github.com/Azure/azure-cli/pull/15363

File: src/azure-cli/azure/cli/command_modules/rdbms/_help.py (hunk 6 above)
Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/_help.py:1246]: will do

File: src/azure-cli/azure/cli/command_modules/rdbms/commands.py (hunk 8 above)
Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/commands.py:503]: This did exist in APIs already but we just added this to CLI just now

Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/commands.py:503]: Will do

**Review by arrownj — APPROVED** [review]
Looks good for `local context` part.

**Review by Juliehzl — APPROVED** [review]
LGTM for command design

**Review by haroldrandom — APPROVED** [review]

**Review by haroldrandom — APPROVED** [review]
will merge once @arde0708 approve

**Review by arde0708 — APPROVED** [review]

**Comment by arde0708** [comment]
> will merge once @arde0708 approve

Done

**Pull request merged**